io/eolearn/io/local_io.py

```diff
@@ -39,18 +42,39 @@ def execute(self, eopatch, *, filename):
         feature_type, feature_name = next(self.feature(eopatch))
         array = eopatch[feature_type][feature_name]
 
-        if self.band_count == 1:
-            array = array[..., 0]
-
         dst_shape = array.shape
-        dst_transform = rasterio.transform.from_bounds(*eopatch.bbox, width=dst_shape[1], height=dst_shape[0])
+        filename_list = []
+        print(self.date_count)
+
+        if str(feature_type) in ['FeatureType.DATA','FeatureType.MASK']:
+            date_timestamp = eopatch.timestamp
+            width = dst_shape[2]
+            height = dst_shape[1]
+            for d in range(self.date_count):
+                for b in range(self.band_count):
+                    date_lyr = array[d,...,b]
+                    print(date_lyr.shape)
+                    filename_list.append(date_lyr)
+
+        else:
+            self.date_count=1
+            width = dst_shape[1]
+            height = dst_shape[0]
+            for b in range(self.band_count):
+                date_lyr = array[...,b]
+                print(date_lyr.shape)
+                filename_list.append(date_lyr)
+
+        dst_transform = rasterio.transform.from_bounds(*eopatch.bbox, width=width, height=height)
         dst_crs = {'init': CRS.ogc_string(eopatch.bbox.crs)}
 
         # Write it out to a file.
         with rasterio.open(os.path.join(self.folder, filename), 'w', driver='GTiff',
-                           width=dst_shape[1], height=dst_shape[0],
-                           count=self.band_count, dtype=self.image_dtype, nodata=self.no_data_value,
+                           width=width, height=height,
+                           count=self.date_count*self.band_count,
+                           dtype=self.image_dtype, nodata=self.no_data_value,
                            transform=dst_transform, crs=dst_crs) as dst:
-            dst.write(array.astype(self.image_dtype), indexes=self.band_count)
+            for id in range(self.date_count*self.band_count):
```

Comment:
All iterations above and this iteration could be handled better. Instead of splitting numpy array into `filename_list` you could do the following:
- Swap array's axis and reshape it into `(height, width, self.date_count * self.band_count)`.
- Rasterize it without this for loop. The code from before wouldn't change much:
```Python
with rasterio.open(os.path.join(self.folder, filename), 'w', driver='GTiff',
                   width=dst_shape[1], height=dst_shape[0],
                   count=self.date_count * self.band_count, dtype=self.image_dtype, nodata=self.no_data_value,
                   transform=dst_transform, crs=dst_crs) as dst:
    dst.write(array.astype(self.image_dtype), indexes=self.date_count * self.band_count)
```
That should also work faster.

Reply:
Would swapping array's axis and reshaping it into `(height, width, self.date_count * self.band_count)` guarantee that the `band` and `time` dimensions are ordered in the correct way (e.g. T1B1 T1B2 T1B3 T2B1 T2B2 T2B3) if `band_count` and `date_count ` are larger than 1?